app/models/user.php

```diff
@@ -122,6 +122,40 @@ class User extends AppModel
         )
     );
 
+    private $defaultSettings = array(
+        'is_public' => false,
+        'lang' => null,
+        'use_most_recent_list' => false,
+        'collapsible_translations' => false,
+        'restrict_search_langs' => false,
+    );
+
+    public function afterFind($results, $primary = false) {
+        foreach ($results as &$result) {
+            if (isset($result['User']) && array_key_exists('settings', $result['User'])) {
```

Comment:
Yea, this is weird... It seems that my PHP version had some bug which evaluated `isset($result['User'])` to true even though `$result` had no `User` key... I updated to another PHP version and it's fine now.